File: solutions/lua/resistor-color-duo/1/resistor-color-duo.lua
local color_codes = {
  black = 0, brown = 1, red = 2, orange = 3, yellow = 4,
  green = 5, blue = 6, violet = 7, grey = 8, white = 9
}
return {
  value = function(colors)

    local first_color = color_codes[colors[ 1 ]]
    local second_color = color_codes[colors[ 2 ]]

    local concat = tostring(first_color) .. tostring(second_color)

    return tonumber(concat)
  end
}

File: solutions/typescript/resistor-color-duo/1/resistor-color-duo.ts
export function decodedValue(colors: string[]) {
  //throw new Error('Remove this line and implement the function')
  var firstColor = Colors[ colors[0] as keyof typeof Colors];
  var secondColor = Colors[ colors[1] as keyof typeof Colors];
  return Number(`${firstColor}${secondColor}`);
}

enum Colors{
  black = 0,
  brown = 1,
  red = 2, 
  orange = 3,
  yellow = 4,
  green = 5,
  blue = 6,
  violet = 7,
  grey = 8,
  white = 9,
}